frame/staking/src/mock.rs

```diff
@@ -612,9 +612,13 @@ pub fn start_session(session_index: SessionIndex) {
 	assert_eq!(Session::current_index(), session_index);
 }
 
+// This start and activate the era given.
+// Because the mock use pallet-session which delays session by one, this will be one session after
+// the election happened, not the first session after the election has happened.
 pub fn start_era(era_index: EraIndex) {
```

Comment:
I modified the test to use current_era instead of active_era but actually the tests generally as you pointed in https://github.com/paritytech/substrate/issues/5244 the tests doesn't test this difference.

Also start_era is ambiguous and used to both start the current era or the active era.
But in the implementation it actually starts the active_era (without taking into account any Forced stuff, maybe would be better to change it with a while loop.